spec/requests/response_templates_spec.rb

```diff
@@ -5,6 +5,97 @@
   let(:moderator) { create(:user, :tag_moderator) }
   let(:admin) { create(:user, :admin) }
 
+  describe "GET /response_templates #index" do
+    it "has status unauthorized if no user is logged in" do
+      get response_templates_path, headers: { HTTP_ACCEPT: "application/json" }
+      expect(response.status_message).to eq "Unauthorized"
+    end
+
+    context "when signed in as a regular user" do
+      before { sign_in user }
+
+      it "responds with JSON" do
+        create(:response_template, user: user, type_of: "personal_comment")
+        get response_templates_path, headers: { HTTP_ACCEPT: "application/json" }
+        expect(response.content_type).to eq "application/json"
+      end
+
+      it "raises RoutingError if the format is not JSON" do
+        expect { get response_templates_path }.to raise_error ActionController::RoutingError
+      end
+
+      it "returns an array of all the user's response templates" do
+        total_response_templates = 2
+        create_list(:response_template, total_response_templates, user: user, type_of: "personal_comment")
+        get response_templates_path, params: { type_of: "personal_comment" }, headers: { HTTP_ACCEPT: "application/json" }
+        parsed_response = JSON.parse(response.body)
```

Comment:
neat little trick I recently learned. `response.parsed_body` will give you the parsed response without you having to parse it yourself manually.